app/models/roster_entry.rb

```diff
@@ -69,18 +69,25 @@ def self.to_csv(user_to_group_map = {})
   # last:   Unlinked student
   #
   # with a secondary sort on ID to ensure ties are always handled in the same way
+  # rubocop:disable Metrics/MethodLength
   def self.order_for_view(assignment)
-    users_with_repo = assignment.repos.pluck(:user_id)
-    sql_formatted_users = users_with_repo.empty? ? "(NULL)" : "(#{users_with_repo.join(',')})"
+    join_sql = <<~SQL
+      LEFT JOIN assignment_repos
+      ON roster_entries.user_id = assignment_repos.user_id
+      AND assignment_repos.assignment_id = #{assignment.id}
+    SQL
 
-    order <<~SQL
+    order_sql = <<~SQL
       CASE
-        WHEN roster_entries.user_id IS NULL THEN 2                   /* Not linked */
-        WHEN roster_entries.user_id IN #{sql_formatted_users} THEN 0 /* Accepted */
-        ELSE 1                                                       /* Linked but not accepted */
+        WHEN roster_entries.user_id IS NULL THEN 2      /* Not linked */
+        WHEN roster_entries.user_id IS NOT NULL THEN 0  /* Linked but not accepted */
```

Comment:
I wonder if there is a spec covering this behavior. I think we may need a small change:

```suggestion
        WHEN roster_entries.user_id IS NOT NULL AND assignment_repos.user_id IS NOT NULL THEN 0  /* Linked but not accepted */
```

Essentially I think this method is attempting to find the three set:

- There is no roster user id
- There is a roster user id and they have a repo
- Everyone else (there is a roster user id and they do not have a repo)

In this case I think the `LEFT JOIN` is a fantastic solution... but we need to actually verify here whether or not the `LEFT JOIN` found a result or not.

Reply:
Thanks for catching that! before this, we only sorted roster_entries into two buckets instead of intended three!